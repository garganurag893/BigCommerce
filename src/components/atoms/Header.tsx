import { FC } from 'react';

interface LinkProps {
  text: string;
  href: string;
}

const Link: FC<LinkProps> = ({ text, href }) => (
  <a
    href={href}
    rel="noreferrer"
    target="_blank"
    className="font-medium text-md mx-3"
  >
    {text}
  </a>
);

const Header = () => (
  <header className="bg-white w-full shadow-md py-2 fixed top-0 z-20">
    <div className="flex flex-row items-center justify-between px-4 sm:px-16">
      <img
        src="/assets/png/logo.png"
        alt="User"
        height={122}
        width={167}
        className="object-contain"
      />
      <nav>
        <Link text="About" href="https://anuraggarg.vercel.app/" />
        <Link
          text="Blogs"
          href="https://medium.com/@garganurag893/how-to-build-game-using-react-native-853455e5b1f5"
        />
        <Link
          text="Special"
          href="https://www.npmjs.com/package/react-native-phone-number-input"
        />
      </nav>
    </div>
  </header>
);

export default Header;
