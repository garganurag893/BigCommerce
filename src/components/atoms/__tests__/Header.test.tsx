import { render } from '@testing-library/react'

import Header from '../Header'

describe('Test Header Component', () => {
  it('Header Snapshot Test', () => {
    const { container } = render(<Header />)
    expect(container).toMatchSnapshot()
  })
})
