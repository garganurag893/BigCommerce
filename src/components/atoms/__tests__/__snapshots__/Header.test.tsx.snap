// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Header Component Header Snapshot Test 1`] = `
<div>
  <header
    class="bg-white w-full shadow-md py-2 fixed top-0 z-20"
  >
    <div
      class="flex flex-row items-center justify-between px-4 sm:px-16"
    >
      <img
        alt="User"
        class="object-contain"
        height="122"
        src="/assets/png/logo.png"
        width="167"
      />
      <nav>
        <a
          class="font-medium text-md mx-3"
          href="https://anuraggarg.vercel.app/"
          rel="noreferrer"
          target="_blank"
        >
          About
        </a>
        <a
          class="font-medium text-md mx-3"
          href="https://medium.com/@garganurag893/how-to-build-game-using-react-native-853455e5b1f5"
          rel="noreferrer"
          target="_blank"
        >
          Blogs
        </a>
        <a
          class="font-medium text-md mx-3"
          href="https://www.npmjs.com/package/react-native-phone-number-input"
          rel="noreferrer"
          target="_blank"
        >
          Special
        </a>
      </nav>
    </div>
  </header>
</div>
`;
